app/src/main/java/ch/epfl/cs311/wanderwave/model/utils/BeaconGenerator.kt

```diff
@@ -0,0 +1,133 @@
+package ch.epfl.cs311.wanderwave.model.utils
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.core.content.ContextCompat
+import ch.epfl.cs311.wanderwave.BuildConfig.MAPS_API_KEY
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import com.google.android.gms.common.api.ApiException
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.net.FindCurrentPlaceRequest
+
+// TODO: What would be a good value for these ?
+private const val MIN_REVIEWS = 100
+private const val MIN_BEACON_DISTANCE = 100.0
+
+// TODO: Store as a resource
+// TODO: Which types do we want ?
+val types: List<String> =
```

Comment:
Good idea, we should discuss in group if some of them are irrelevant or if we should add more (if possible)